cmd/dep/gopath_scanner.go

```diff
@@ -25,17 +25,19 @@ type gopathScanner struct {
 	ctx        *dep.Ctx
 	directDeps map[string]bool
 	sm         gps.SourceManager
+	gopath     bool
 
 	pd    projectData
 	origM *dep.Manifest
 	origL *dep.Lock
 }
 
-func newGopathScanner(ctx *dep.Ctx, directDeps map[string]bool, sm gps.SourceManager) *gopathScanner {
+func newGopathScanner(ctx *dep.Ctx, directDeps map[string]bool, sm gps.SourceManager, gopath bool) *gopathScanner {
```

Comment:
I think this went a bit too far in trying to keep init "clean". 😀  Let's have the init command handle the `-gopath` flag and call `gopathScanner` or not based on its value.